pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1750,3 +1770,85 @@ func (r *ReconcileClusterDeployment) deleteStaleProvisions(provs []*hivev1.Clust
 		}
 	}
 }
+
+// getAllSyncSetInstances return all syncset instances for a cluster deployment
+func (r *ReconcileClusterDeployment) getAllSyncSetInstances(cd *hivev1.ClusterDeployment) ([]*hivev1.SyncSetInstance, error) {
+
+	list := &hivev1.SyncSetInstanceList{}
+	err := r.List(context.TODO(), list, client.InNamespace(cd.Namespace))
+	if err != nil {
+		return nil, err
+	}
+
+	syncSetInstances := []*hivev1.SyncSetInstance{}
+	for i, syncSetInstance := range list.Items {
+		fmt.Println("Temp: syncsetinstance name: ", syncSetInstance.Name)
+		if syncSetInstance.Spec.ClusterDeployment.Name == cd.Name {
+			syncSetInstances = append(syncSetInstances, &list.Items[i])
+		}
+	}
+	return syncSetInstances, nil
+}
+
+// checkFailureSyncCondition return true when syncsetinstance status condition has hivev1.ApplyFailureSyncCondition
+func checkFailureSyncCondition(syncSetInstances []*hivev1.SyncSetInstance) bool {
+	for _, syncSetInstance := range syncSetInstances {
+		for _, condition := range syncSetInstance.Status.Conditions {
```

Comment:
Need to search through the conditions for each of the resources, patches, and secret references. Also, need to look for `UnknownObjectSyncCondition` and `DeletionFailedSyncCondition` conditions types, too.

Reply:
Trying to understand  `UnknownObjectSyncCondition` and `DeletionFailedSyncCondition` . Thanks

Reply:
I think `UnknownObjectSyncCondition` should be part of status condistions which we are going through, refer  https://github.com/openshift/hive/blob/master/pkg/controller/syncsetinstance/syncsetinstance_controller.go#L508

Reply:
`DeletionFailedSyncCondition` as it is part of  `resources` https://github.com/openshift/hive/blob/master/pkg/controller/syncsetinstance/syncsetinstance_controller.go#L395-L423